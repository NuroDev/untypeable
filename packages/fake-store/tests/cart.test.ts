import { describe, it, expect } from "vitest";

import {
  CartSchema,
  CartsSchema,
  UpdatedCartSchema,
  UserCartsSchema,
} from "../src/zod";
import { useTestClient } from "./_client";

describe("fakestoreapi.com - cart", () => {
  const client = useTestClient();

  describe("DELETE", () => {
    it("/carts/:id", async () => {
      const deletedCart = await client("DELETE", "/carts/:id", {
        id: 1,
      });

      expect(deletedCart).toMatchSnapshot();
      expect(CartSchema.safeParse(deletedCart).success).toBe(true);
    });
  });

  describe("GET", () => {
    it("/carts", async () => {
      const carts = await client("GET", "/carts");

      expect(carts).toMatchSnapshot();
      expect(CartsSchema.safeParse(carts).success).toBe(true);
    });

    it("/carts?limit=2", async () => {
      const limitedCarts = await client("GET", "/carts", { limit: 2 });

      expect(limitedCarts).toMatchSnapshot();
      expect(limitedCarts).toHaveLength(2);
      expect(CartsSchema.safeParse(limitedCarts).success).toBe(true);
    });

    it("/carts?order=desc", async () => {
      const orderedCarts = await client("GET", "/carts", {
        order: "desc",
      });

      expect(orderedCarts).toMatchSnapshot();
      expect(CartsSchema.safeParse(orderedCarts).success).toBe(true);
    });

    it("/carts?startdate=2019-12-10&enddate=2020-10-10", async () => {
      const dateRangeCarts = await client("GET", "/carts", {
        startdate: "2019-12-10",
        enddate: "2020-10-10",
      });

      expect(dateRangeCarts).toMatchSnapshot();
      expect(CartsSchema.safeParse(dateRangeCarts).success).toBe(true);
    });

    it("/carts?limit=1&order=desc", async () => {
      const carts = await client("GET", "/carts", {
        limit: 1,
        order: "desc",
      });

      expect(carts).toMatchSnapshot();
      expect(carts).toHaveLength(1);
      expect(CartsSchema.safeParse(carts).success).toBe(true);
    });

    it("/carts/:id", async () => {
      const cart = await client("GET", "/carts/:id", {
        id: 1,
      });

      expect(cart).toMatchSnapshot();
      expect(CartSchema.safeParse(cart).success).toBe(true);
    });

    it("/carts/user/:userId", async () => {
      const userCarts = await client("GET", "/carts/user/:userId", {
        userId: 2,
      });

      expect(userCarts).toMatchSnapshot();
      expect(UserCartsSchema.safeParse(userCarts).success).toBe(true);
    });
  });

  describe("PATCH", () => {
    it("/products/:id", async () => {
      const updatedCart = await client("PATCH", "/carts/:id", {
        date: new Date(0),
        id: 1,
        products: [],
        userId: 3,
      });

      expect(updatedCart).toMatchSnapshot();
      expect(CartSchema.safeParse(updatedCart).success).toBe(true);
    });
  });

  describe("POST", () => {
    it("/products", async () => {
      const newCart = await client("POST", "/carts", {
        date: new Date(0),
        products: [],
        userId: 3,
      });

      expect(newCart).toMatchSnapshot();
      expect(CartSchema.safeParse(newCart).success).toBe(true);
    });
  });

  describe("PUT", () => {
    it("/carts/:id", async () => {
      const updatedCart = await client("PUT", "/carts/:id", {
        products: [],
        userId: 3,
        id: 1,
      });

      expect(updatedCart).toMatchSnapshot();
      expect(UpdatedCartSchema.safeParse(updatedCart).success).toBe(true);
    });
  });
});
