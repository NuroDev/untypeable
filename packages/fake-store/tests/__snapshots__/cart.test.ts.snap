// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fakestoreapi.com - cart > DELETE > /carts/:id 1`] = `
{
  "__v": 0,
  "date": "2020-03-02T00:00:00.000Z",
  "id": 1,
  "products": [
    {
      "productId": 1,
      "quantity": 4,
    },
    {
      "productId": 2,
      "quantity": 1,
    },
    {
      "productId": 3,
      "quantity": 6,
    },
  ],
  "userId": 1,
}
`;

exports[`fakestoreapi.com - cart > GET > /carts 1`] = `
[
  {
    "__v": 0,
    "date": "2020-03-02T00:00:00.000Z",
    "id": 1,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
      {
        "productId": 2,
        "quantity": 1,
      },
      {
        "productId": 3,
        "quantity": 6,
      },
    ],
    "userId": 1,
  },
  {
    "__v": 0,
    "date": "2020-01-02T00:00:00.000Z",
    "id": 2,
    "products": [
      {
        "productId": 2,
        "quantity": 4,
      },
      {
        "productId": 1,
        "quantity": 10,
      },
      {
        "productId": 5,
        "quantity": 2,
      },
    ],
    "userId": 1,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 3,
    "products": [
      {
        "productId": 1,
        "quantity": 2,
      },
      {
        "productId": 9,
        "quantity": 1,
      },
    ],
    "userId": 2,
  },
  {
    "__v": 0,
    "date": "2020-01-01T00:00:00.000Z",
    "id": 4,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
    ],
    "userId": 3,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 5,
    "products": [
      {
        "productId": 7,
        "quantity": 1,
      },
      {
        "productId": 8,
        "quantity": 1,
      },
    ],
    "userId": 3,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 6,
    "products": [
      {
        "productId": 10,
        "quantity": 2,
      },
      {
        "productId": 12,
        "quantity": 3,
      },
    ],
    "userId": 4,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 7,
    "products": [
      {
        "productId": 18,
        "quantity": 1,
      },
    ],
    "userId": 8,
  },
]
`;

exports[`fakestoreapi.com - cart > GET > /carts/:id 1`] = `
{
  "__v": 0,
  "date": "2020-03-02T00:00:00.000Z",
  "id": 1,
  "products": [
    {
      "productId": 1,
      "quantity": 4,
    },
    {
      "productId": 2,
      "quantity": 1,
    },
    {
      "productId": 3,
      "quantity": 6,
    },
  ],
  "userId": 1,
}
`;

exports[`fakestoreapi.com - cart > GET > /carts/user/:userId 1`] = `
[
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 3,
    "products": [
      {
        "productId": 1,
        "quantity": 2,
      },
      {
        "productId": 9,
        "quantity": 1,
      },
    ],
    "userId": 2,
  },
]
`;

exports[`fakestoreapi.com - cart > GET > /carts?limit=1&order=desc 1`] = `
[
  {
    "__v": 0,
    "date": "2020-03-02T00:00:00.000Z",
    "id": 1,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
      {
        "productId": 2,
        "quantity": 1,
      },
      {
        "productId": 3,
        "quantity": 6,
      },
    ],
    "userId": 1,
  },
]
`;

exports[`fakestoreapi.com - cart > GET > /carts?limit=2 1`] = `
[
  {
    "__v": 0,
    "date": "2020-03-02T00:00:00.000Z",
    "id": 1,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
      {
        "productId": 2,
        "quantity": 1,
      },
      {
        "productId": 3,
        "quantity": 6,
      },
    ],
    "userId": 1,
  },
  {
    "__v": 0,
    "date": "2020-01-02T00:00:00.000Z",
    "id": 2,
    "products": [
      {
        "productId": 2,
        "quantity": 4,
      },
      {
        "productId": 1,
        "quantity": 10,
      },
      {
        "productId": 5,
        "quantity": 2,
      },
    ],
    "userId": 1,
  },
]
`;

exports[`fakestoreapi.com - cart > GET > /carts?order=desc 1`] = `
[
  {
    "__v": 0,
    "date": "2020-03-02T00:00:00.000Z",
    "id": 1,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
      {
        "productId": 2,
        "quantity": 1,
      },
      {
        "productId": 3,
        "quantity": 6,
      },
    ],
    "userId": 1,
  },
  {
    "__v": 0,
    "date": "2020-01-02T00:00:00.000Z",
    "id": 2,
    "products": [
      {
        "productId": 2,
        "quantity": 4,
      },
      {
        "productId": 1,
        "quantity": 10,
      },
      {
        "productId": 5,
        "quantity": 2,
      },
    ],
    "userId": 1,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 3,
    "products": [
      {
        "productId": 1,
        "quantity": 2,
      },
      {
        "productId": 9,
        "quantity": 1,
      },
    ],
    "userId": 2,
  },
  {
    "__v": 0,
    "date": "2020-01-01T00:00:00.000Z",
    "id": 4,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
    ],
    "userId": 3,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 5,
    "products": [
      {
        "productId": 7,
        "quantity": 1,
      },
      {
        "productId": 8,
        "quantity": 1,
      },
    ],
    "userId": 3,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 6,
    "products": [
      {
        "productId": 10,
        "quantity": 2,
      },
      {
        "productId": 12,
        "quantity": 3,
      },
    ],
    "userId": 4,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 7,
    "products": [
      {
        "productId": 18,
        "quantity": 1,
      },
    ],
    "userId": 8,
  },
]
`;

exports[`fakestoreapi.com - cart > GET > /carts?startdate=2019-12-10&enddate=2020-10-10 1`] = `
[
  {
    "__v": 0,
    "date": "2020-03-02T00:00:00.000Z",
    "id": 1,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
      {
        "productId": 2,
        "quantity": 1,
      },
      {
        "productId": 3,
        "quantity": 6,
      },
    ],
    "userId": 1,
  },
  {
    "__v": 0,
    "date": "2020-01-02T00:00:00.000Z",
    "id": 2,
    "products": [
      {
        "productId": 2,
        "quantity": 4,
      },
      {
        "productId": 1,
        "quantity": 10,
      },
      {
        "productId": 5,
        "quantity": 2,
      },
    ],
    "userId": 1,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 3,
    "products": [
      {
        "productId": 1,
        "quantity": 2,
      },
      {
        "productId": 9,
        "quantity": 1,
      },
    ],
    "userId": 2,
  },
  {
    "__v": 0,
    "date": "2020-01-01T00:00:00.000Z",
    "id": 4,
    "products": [
      {
        "productId": 1,
        "quantity": 4,
      },
    ],
    "userId": 3,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 5,
    "products": [
      {
        "productId": 7,
        "quantity": 1,
      },
      {
        "productId": 8,
        "quantity": 1,
      },
    ],
    "userId": 3,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 6,
    "products": [
      {
        "productId": 10,
        "quantity": 2,
      },
      {
        "productId": 12,
        "quantity": 3,
      },
    ],
    "userId": 4,
  },
  {
    "__v": 0,
    "date": "2020-03-01T00:00:00.000Z",
    "id": 7,
    "products": [
      {
        "productId": 18,
        "quantity": 1,
      },
    ],
    "userId": 8,
  },
]
`;

exports[`fakestoreapi.com - cart > PATCH > /products/:id 1`] = `
{
  "date": "1970-01-01T00:00:00.000Z",
  "id": 1,
  "products": [],
  "userId": 3,
}
`;

exports[`fakestoreapi.com - cart > POST > /products 1`] = `
{
  "date": "1970-01-01T00:00:00.000Z",
  "id": 11,
  "products": [],
  "userId": 3,
}
`;

exports[`fakestoreapi.com - cart > PUT > /carts/:id 1`] = `
{
  "id": 1,
  "products": [],
  "userId": 3,
}
`;
